src/object/endpoint.c

```diff
@@ -507,3 +507,25 @@ void reorderEP(endpoint_t *epptr, tcb_t *thread)
     ep_ptr_set_queue(epptr, queue);
 }
 #endif
+
+
+#ifdef CONFIG_KERNEL_MCS
+void setThreshold(endpoint_t * epptr, time_t threshold, bool_t budgetLimit) {
+    /* Add the kernel WCET to the passed threshold value, unless we were passed 0*/
```

Comment:
The comment is in the wrong place, but why add WCET? A fastpath invocation should take nowhere near WCET time.

I'd assure that the threshold is at least WCET instead. Users need to measure the call time to tweak this setting anyway, don't make it harder by changing the value randomly.